linera-witty/src/wit_generation/stub_instance.rs

```diff
@@ -0,0 +1,116 @@
+// Copyright (c) Zefchain Labs, Inc.
+// SPDX-License-Identifier: Apache-2.0
+
+//! Contains a [`StubInstance`] type and the code it requires to implement the necessary
+//! trait to be used as a Wasm instance type during compile time.
+
+use std::{borrow::Cow, marker::PhantomData};
+
+use crate::{
+    memory_layout::FlatLayout, GuestPointer, Instance, InstanceWithFunction, InstanceWithMemory,
+    Runtime, RuntimeError, RuntimeMemory,
+};
+
+/// A stub Wasm instance.
+///
+/// This is when a Wasm instance type is needed for type-checking but is not needed during
+/// runtime.
+pub struct StubInstance<UserData = ()> {
+    _user_data: PhantomData<UserData>,
+}
+
+impl<UserData> Default for StubInstance<UserData> {
+    fn default() -> Self {
+        StubInstance {
+            _user_data: PhantomData,
+        }
+    }
+}
+
+impl<UserData> Instance for StubInstance<UserData> {
+    type Runtime = StubRuntime;
+    type UserData = UserData;
+    type UserDataReference<'a> = &'a UserData
+    where
+        Self::UserData: 'a,
+        Self: 'a
+        ;
```

Comment:
(Did `rustfmt` want that?)

Reply:
No, I think I expected `rustfmt` to fix it but it didn't :(